### ISSUE TITLE
x/website: Invalid compiler requirements for race detector on Windows

### ISSUE THREAD

**Issue by antoninbas** [golang/go#61916]
The current [documentation](https://go.dev/doc/articles/race_detector) reads:

> On Windows, the race detector runtime is sensitive to the version of the C compiler installed; as of Go 1.21, building a program with -race requires a C compiler that incorporates version 8 or later of the mingw-w64 runtime libraries. You can test your C compiler by invoking it with the arguments --print-file-name libsynchronization.a. A newer compliant C compiler will print a full path for this library, whereas older C compilers will just echo the argument.

According to the documentation, version 8 of the mingw-w64 runtime libraries should work, as it is the minimum required version ("version 8 or later"). However, it doesn't seem to work.

I have tested using Github Actions runners for Windows. Github actions supports 2 runners: `windows-2019` (which comes with `Mingw-w64 8.1.0`), and `windows-2022` (which comes with `Mingw-w64 11.2.0`). I can run the race detector successfully with the latter (`windows-2022`), but not with the former (`windows-2019`). When using ``windows-2019` and `Mingw-w64 8.1.0`, the observed error is the same as for https://github.com/golang/go/issues/61058: `exit status 0xc0000139`.

`windows-2019` runner:
```
> gcc --print-file-name libsynchronization.a
C:/ProgramData/Chocolatey/lib/mingw/tools/install/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.1.0/../../../../x86_64-w64-mingw32/lib/../lib/libsynchronization.a
```

`windows-2022` runner:
```
Run gcc --print-file-name libsynchronization.a
c:/programdata/chocolatey/lib/mingw/tools/install/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/lib/../lib/libsynchronization.a
```

I don't know what the exact requirement is, but `8.1.0` does not work (unless some other dependency is at fault).

Installed software on Github Actions Windows runners:
- [Windows 2019](https://github.com/actions/runner-images/blob/main/images/win/Windows2019-Readme.md)
- [Windows 2022](https://github.com/actions/runner-images/blob/main/images/win/Windows2022-Readme.md)

Note that I am using Go v1.21, and that I didn't have the same issue with Go v1.19.

**Comment by ChristoWolf** [comment]
Hi!

Just wanted to chime in that I am experiencing the same issue with `mingw-w64-8.1.0`.

**Comment by qmuntal** [comment]
I've reproduced this issue by using the same Mingw-w64 installion used in the `windows-2019` GitHub Actions image. I can confirm that the issue is the same as this one: https://github.com/golang/go/issues/61058#issuecomment-1614686520. For whatever reason that Mingw-w64 distribution doesn't contain the following upstream Mingw-w64 fix: https://github.com/mingw-w64/mingw-w64/commit/68d0e7d551d2e2a5b8a15bdca3f8d15fc097ed01.

Closing as duplicated.
